src/Microsoft.OData.Core/ODataAsynchronousWriter.cs

```diff
@@ -102,9 +104,13 @@ void IODataOutputInStreamErrorListener.OnInStreamError()
         }
 
         /// <inheritdoc/>
-        Task IODataOutputInStreamErrorListener.OnInStreamErrorAsync()
+        async Task IODataOutputInStreamErrorListener.OnInStreamErrorAsync()
         {
-            throw new NotImplementedException();
+            this.rawOutputContext.VerifyNotDisposed();
+            await this.rawOutputContext.TextWriter.FlushAsync()

```

Comment:
Do we need to flush if we are still going to throw the exception below?

Reply:
Is the rawOutputContext reused?

Reply:
@marabooy In particular scenarios, we return the partial payload written before things went haywire. Flushing is therefore necessary. We're not flushing so we can reuse the `RawOutputContext` instance.